typescript/lib/networks/network_context.ts

```diff
@@ -0,0 +1,628 @@
+        // }
+
+        const promiseAsyncLoadLambda: Promise<ResultAsync<any, Error>> = new Promise((resolve, _unusedReject) => { //TODO - Repeating this promise format a lot, make a function for it?
+            this.client.executeLambda(args, (_unusedError: grpc.ServiceError, response: any) => {
+                resolve(okAsync(response));
+            })
+        });
+        const promiseLoadLambda: Result<any, Error> = await promiseAsyncLoadLambda;
+        if (!promiseLoadLambda.isOk()) {
+            return [null, promiseLoadLambda.error];
+        }
+
+        const moduleCtx: LambdaContext = new LambdaContext(this.client, lambdaId);
+        return [moduleCtx, null];
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public async getLambdaContext(lambdaId: LambdaID): Promise<[LambdaContext, Error]> {
+        const args: GetLambdaInfoArgs = newGetLambdaInfoArgs(lambdaId);
+        
+        // TODO TODO TODO - CALLBACK & ERROR-HANDLING (REMOVE)
+        // // NOTE: As of 2021-07-18, we actually don't use any of the info that comes back because the LambdaContext doesn't require it!
+        // _, err := networkCtx.client.GetLambdaInfo(context.Background(), args)
+        // if err !== nil {
+        //     return nil, stacktrace.Propagate(err, "An error occurred getting info for Lambda '%v'", lambdaId)
+        // }
+
+        const promiseAsyncGetLambdaInfo: Promise<ResultAsync<GetLambdaInfoResponse, Error>> = new Promise((resolve, _unusedReject) => {
+            this.client.getLambdaInfo(args, (_unusedError: grpc.ServiceError, response: GetLambdaInfoResponse) => {
+                resolve(okAsync(response));
+            })
+        });
+        const promiseGetLambdaInfo: Result<GetLambdaInfoResponse, Error> = await promiseAsyncGetLambdaInfo;
+        if (!promiseGetLambdaInfo.isOk()) {
+            return [null, promiseGetLambdaInfo.error];
+        }
+
+        const lambdaCtx: LambdaContext = new LambdaContext(this.client, lambdaId);
+        return [lambdaCtx, null];
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public async registerStaticFiles(staticFileFilepaths: Map<StaticFileID, string>): Promise<Error> {
+        const strSet: Map<string, boolean> = new Map();
+        for (let [staticFileId, srcAbsFilepath] of staticFileFilepaths.entries()) {
+            
+            // Sanity-check that the source filepath exists
+            const promiseAsyncSrcAbsFilepath: Promise<ResultAsync<fs.Stats, Error>> = new Promise((resolve, _unusedReject) => {
+                fs.stat(srcAbsFilepath, (_unusedError: Error, response: fs.Stats) => { //TODO - should I be using fsPromises.stat() instead, I personally want to remain consistent with manually constructing promises
```

Comment:
So manually constructing promises is a last resort, that we're only doing because we have to. Whenever someone does it for us, we want to use that (because it's less code we have to write so therefore less opportunity for bugs).

E.g. this actually suffers from the same problem as the other handwritten promises, where the `Error` isn't checked (even though it should be)